<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read data — read • miplicorn</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Read data — read" />
<meta property="og:description" content="Read files containing
MIPtools' data tables.
read_file() reads a single file. read() is a convenience function that
reads all files output by
MIPtools and combines them.
Data files include the reference table, the alternate table, and the coverage
table. Data is read lazily using the
vroom package. Data can be
filtered, retaining all rows that satisfy the conditions. To be retained, the
row in question must produce a value of TRUE for all conditions. Note that
when a condition evaluates to NA, the row will be dropped." />
<meta property="og:image" content="https://bailey-lab.github.io/miplicorn/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">miplicorn</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/miplicorn.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/large-files.html">Large data sets</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../news/index.html">Change log</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bailey-lab/miplicorn/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read data</h1>
    <small class="dont-index">Source: <a href='https://github.com/bailey-lab/miplicorn/blob/master/R/read.R'><code>R/read.R</code></a></small>
    <div class="hidden name"><code>read.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read files containing
<a href='https://github.com/bailey-lab/MIPTools'>MIPtools'</a> data tables.
<code>read_file()</code> reads a single file. <code>read()</code> is a convenience function that
reads all files output by
<a href='https://github.com/bailey-lab/MIPTools'>MIPtools</a> and combines them.
Data files include the reference table, the alternate table, and the coverage
table. Data is read lazily using the
<a href='https://vroom.r-lib.org/index.html'><code>vroom</code></a> package. Data can be
filtered, retaining all rows that satisfy the conditions. To be retained, the
row in question must produce a value of <code>TRUE</code> for all conditions. Note that
when a condition evaluates to NA, the row will be dropped.</p>
    </div>

    <pre class="usage"><span class='fu'>read</span><span class='op'>(</span>
  <span class='va'>.ref_file</span>,
  <span class='va'>.alt_file</span>,
  <span class='va'>.cov_file</span>,
  <span class='va'>...</span>,
  chrom <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Deprecated.html'>deprecated</a></span><span class='op'>(</span><span class='op'>)</span>,
  gene <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Deprecated.html'>deprecated</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span>

<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>.file</span>, <span class='va'>...</span>, .name <span class='op'>=</span> <span class='st'>"value"</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>.ref_file</th>
      <td><p>File path to the reference table.</p></td>
    </tr>
    <tr>
      <th>.alt_file</th>
      <td><p>File path to the alternate table.</p></td>
    </tr>
    <tr>
      <th>.cov_file</th>
      <td><p>File path to the coverage table.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>&lt;<code>data-masking</code>&gt; Expressions that return a
logical value and are used to filter the data. If multiple expressions are
included, they are combined with the <code>&amp;</code> operator. Only rows for which all
conditions evaluate to <code>TRUE</code> are kept.</p></td>
    </tr>
    <tr>
      <th>chrom</th>
      <td><p><a href='https://lifecycle.r-lib.org/articles/stages.html#deprecated'><img src='figures/lifecycle-deprecated.svg' alt='[Deprecated]' /></a>: The chromosome(s) to filter
to.</p></td>
    </tr>
    <tr>
      <th>gene</th>
      <td><p><a href='https://lifecycle.r-lib.org/articles/stages.html#deprecated'><img src='figures/lifecycle-deprecated.svg' alt='[Deprecated]' /></a>: The gene(s) to filter to.</p></td>
    </tr>
    <tr>
      <th>.file</th>
      <td><p>File path to a file.</p></td>
    </tr>
    <tr>
      <th>.name</th>
      <td><p>The information contained in the specific file. For example
<code>"coverage"</code> or <code>"ref_umi_count"</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>tibble()</code>. The first six columns contain the metadata associated
with each sample and mutation. Columns <code>ref_umi_count</code> and <code>alt_umi_count</code>
contain the umi count of the reference and alternate allele, respectively.
Column <code>coverage</code> contains the coverage for each data point.</p>
    <h2 class="hasAnchor" id="data-structure"><a class="anchor" href="#data-structure"></a>Data structure</h2>

    

<p>Input data must contain six rows of metadata. The metadata can vary depending
on what type of file is read, but typically contains information about the
location of a mutation. The remaining rows represent the data for each sample
sequenced. Together, the alternate, reference, and coverage tables can
provide information about mutations observed and the coverage at each site.</p>
    <h2 class="hasAnchor" id="useful-filter-functions"><a class="anchor" href="#useful-filter-functions"></a>Useful filter functions</h2>

    

<p>The <code><a href='https://dplyr.tidyverse.org/reference/filter.html'>dplyr::filter()</a></code> function is employed to subset the rows of the data
applying the expressions in <code>...</code> to the column values to determine which
rows should be retained.</p>
<p>There are many functions and operators that are useful when constructing the
expressions used to filter the data:</p><ul>
<li><p><code><a href='https://rdrr.io/r/base/Comparison.html'>==</a></code>, <code>&gt;</code>, <code>&gt;=</code>, etc.</p></li>
<li><p><code>&amp;</code>, <code><a href='https://rdrr.io/r/base/Logic.html'>|</a></code>, <code><a href='https://rdrr.io/r/base/Logic.html'>!</a></code>, <code><a href='https://rdrr.io/r/base/Logic.html'>xor()</a></code></p></li>
<li><p><code><a href='https://rdrr.io/r/base/NA.html'>is.na()</a></code></p></li>
<li><p><code>between()</code>, <code>near()</code></p></li>
</ul>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://vroom.r-lib.org/reference/vroom.html'>vroom::vroom()</a></code> <code><a href='https://dplyr.tidyverse.org/reference/filter.html'>dplyr::filter()</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Get path to example file</span>
<span class='va'>ref_file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='miplicorn_example.html'>miplicorn_example</a></span><span class='op'>(</span><span class='st'>"reference_AA_table.csv"</span><span class='op'>)</span>
<span class='va'>alt_file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='miplicorn_example.html'>miplicorn_example</a></span><span class='op'>(</span><span class='st'>"alternate_AA_table.csv"</span><span class='op'>)</span>
<span class='va'>cov_file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='miplicorn_example.html'>miplicorn_example</a></span><span class='op'>(</span><span class='st'>"coverage_AA_table.csv"</span><span class='op'>)</span>
<span class='va'>ref_file</span>
</div><div class='output co'>#&gt; [1] "/home/runner/work/_temp/Library/miplicorn/extdata/reference_AA_table.csv"</div><div class='input'>
<span class='co'># Input sources -------------------------------------------------------------</span>
<span class='co'># Read from a path</span>
<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 6,344 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 6,334 more rows</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 6,344 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 6,334 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div><div class='input'>
<span class='co'># You can also use paths directly</span>
<span class='co'># read_file("reference_AA_table.csv")</span>
<span class='co'># read("reference_AA_table.csv", "alternate_AA_table.csv", "coverage_AA_table.csv")</span>

<span class='co'># Read entire file ----------------------------------------------------------</span>
<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 6,344 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 6,334 more rows</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 6,344 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 6,334 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div><div class='input'>
<span class='co'># Data filtering ------------------------------------------------------------</span>
<span class='co'># Filtering by one criterion</span>
<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 260 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 250 more rows</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 260 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 250 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div><div class='input'>
<span class='co'># Filtering by multiple criteria within a single logical expression</span>
<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span> <span class='op'>&amp;</span> <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 156 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 146 more rows</span></div><div class='input'><span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span> <span class='op'>|</span> <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 2,496 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 2,486 more rows</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span> <span class='op'>&amp;</span> <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 156 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 146 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span> <span class='op'>|</span> <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 2,496 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 2,486 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div><div class='input'>
<span class='co'># When multiple expressions are used, they are combined using &amp;</span>
<span class='fu'>read_file</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span>, <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span>, .name <span class='op'>=</span> <span class='st'>"umi"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 156 × 8</span>
#&gt;    sample     gene_id  gene  mutation_name  exonic_func aa_change targeted   umi
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span>
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        608
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         20
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        158
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          2
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          1
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        129
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes          0
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes         90
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0… atp6  atp6-Ala623Glu missense_v… Ala623Glu Yes        175
#&gt; <span style='color: #949494;'># … with 146 more rows</span></div><div class='input'><span class='fu'>read</span><span class='op'>(</span><span class='va'>ref_file</span>, <span class='va'>alt_file</span>, <span class='va'>cov_file</span>, <span class='va'>gene</span> <span class='op'>==</span> <span class='st'>"atp6"</span>, <span class='va'>targeted</span> <span class='op'>==</span> <span class='st'>"Yes"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 156 × 10</span>
#&gt;    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   
#&gt; <span style='color: #BCBCBC;'> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #BCBCBC;'>10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#&gt; <span style='color: #949494;'># … with 146 more rows, and 3 more variables: ref_umi_count &lt;dbl&gt;,</span>
#&gt; <span style='color: #949494;'>#   alt_umit_count &lt;dbl&gt;, coverage &lt;dbl&gt;</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Aris Paschalidis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


