<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="The read_tbl_*() family of functions is designed to read data tables
generated by the software program
MIPTools. Data is read lazily
using the vroom package. Data can be
filtered, retaining all rows that satisfy the conditions. To be retained, the
row in question must produce a value of TRUE for all conditions. Note that
when a condition evaluates to NA, the row will be dropped."><title>Read MIPTools tables — read-tbl • miplicorn</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Lato-0.4.2/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.2/font.css" rel="stylesheet"><link href="../deps/Fira_Sans-0.4.2/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Read MIPTools tables — read-tbl"><meta property="og:description" content="The read_tbl_*() family of functions is designed to read data tables
generated by the software program
MIPTools. Data is read lazily
using the vroom package. Data can be
filtered, retaining all rows that satisfy the conditions. To be retained, the
row in question must produce a value of TRUE for all conditions. Note that
when a condition evaluates to NA, the row will be dropped."><meta property="og:image" content="https://bailey-lab.github.io/miplicorn/logo.png"><meta name="robots" content="noindex"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">miplicorn</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.2.1.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/miplicorn.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/large-files.html">Dealing with large data sets</a>
    <a class="dropdown-item" href="../articles/faq.html">FAQ</a>
  </div>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-news">News</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-news">
    <a class="dropdown-item" href="../news/index.html">Change log</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/bailey-lab/miplicorn/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Read MIPTools tables</h1>
      <small class="dont-index">Source: <a href="https://github.com/bailey-lab/miplicorn/blob/HEAD/R/read-tbl.R" class="external-link"><code>R/read-tbl.R</code></a></small>
      <div class="d-none name"><code>read-tbl.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The <code>read_tbl_*()</code> family of functions is designed to read data tables
generated by the software program
<a href="https://github.com/bailey-lab/MIPTools" class="external-link"><code>MIPTools</code></a>. Data is read lazily
using the <a href="https://vroom.r-lib.org/index.html" class="external-link"><code>vroom</code> package</a>. Data can be
filtered, retaining all rows that satisfy the conditions. To be retained, the
row in question must produce a value of <code>TRUE</code> for all conditions. Note that
when a condition evaluates to NA, the row will be dropped.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">read_tbl_reference</span><span class="op">(</span><span class="va">.tbl</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">read_tbl_alternate</span><span class="op">(</span><span class="va">.tbl</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">read_tbl_coverage</span><span class="op">(</span><span class="va">.tbl</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">read_tbl_genotype</span><span class="op">(</span><span class="va">.tbl</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">read_tbl_haplotype</span><span class="op">(</span><span class="va">.tbl</span>, <span class="va">...</span>, .col_select <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">read_tbl_ref_alt_cov</span><span class="op">(</span></span>
<span>  <span class="va">.tbl_ref</span>,</span>
<span>  <span class="va">.tbl_alt</span>,</span>
<span>  <span class="va">.tbl_cov</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  chrom <span class="op">=</span> <span class="fu">deprecated</span><span class="op">(</span><span class="op">)</span>,</span>
<span>  gene <span class="op">=</span> <span class="fu">deprecated</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>.tbl</dt>
<dd><p>File path to the table.</p></dd>


<dt>...</dt>
<dd><p>&lt;<code><a href="https://dplyr.tidyverse.org/reference/dplyr_data_masking.html" class="external-link">data-masking</a></code>&gt; Filtering
expressions. Expressions must return a logical value. If multiple
expressions are included, they are combined with the <code>&amp;</code> operator. Only
rows for which all conditions evaluate to <code>TRUE</code> are kept.</p></dd>


<dt>.col_select</dt>
<dd><p>Columns to include in the results. Columns can be selected
using one or more selection expressions as in
<code><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">dplyr::select()</a></code>. Use <code><a href="https://rdrr.io/r/base/c.html" class="external-link">c()</a></code> or <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code> to use more
than one expression. See <code><a href="https://tidyselect.r-lib.org/reference/language.html" class="external-link">?tidyselect::language</a></code>
for details on available selection options.</p></dd>


<dt>.tbl_ref</dt>
<dd><p>File path to the reference table.</p></dd>


<dt>.tbl_alt</dt>
<dd><p>File path to the alternate table.</p></dd>


<dt>.tbl_cov</dt>
<dd><p>File path to the coverage table.</p></dd>


<dt>chrom</dt>
<dd><p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated" class="external-link"><img src="figures/lifecycle-deprecated.svg" alt="[Deprecated]"></a> The chromosome(s) to filter
to.</p></dd>


<dt>gene</dt>
<dd><p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated" class="external-link"><img src="figures/lifecycle-deprecated.svg" alt="[Deprecated]"></a> The gene(s) to filter to.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble()</a></code>. The first six columns contain the
metadata associated with each sample and mutation. The last column contains
the information parsed from the table. In some cases, this may be the
umi_count and in other cases it may be the coverage of the associated data
point.</p>
    </div>
    <div class="section level2">
    <h2 id="data-structure">Data structure<a class="anchor" aria-label="anchor" href="#data-structure"></a></h2>
    

<p>Input data must contain six rows of metadata. The metadata can vary depending
on what type of file is read, but typically contains information about the
location of a mutation. The remaining rows represent the data for each sample
sequenced.</p>
    </div>
    <div class="section level2">
    <h2 id="useful-filter-functions">Useful filter functions<a class="anchor" aria-label="anchor" href="#useful-filter-functions"></a></h2>
    

<p>The <code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter()</a></code> function is employed to subset the rows of the data
applying the expressions in <code>...</code> to the column values to determine which
rows should be retained.</p>
<p>There are many functions and operators that are useful when constructing the
expressions used to filter the data:</p><ul><li><p><code><a href="https://rdrr.io/r/base/Comparison.html" class="external-link">==</a></code>, <code>&gt;</code>, <code>&gt;=</code>, etc.</p></li>
<li><p><code>&amp;</code>, <code><a href="https://rdrr.io/r/base/Logic.html" class="external-link">|</a></code>, <code><a href="https://rdrr.io/r/base/Logic.html" class="external-link">!</a></code>, <code><a href="https://rdrr.io/r/base/Logic.html" class="external-link">xor()</a></code></p></li>
<li><p><code><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na()</a></code></p></li>
<li><p><code><a href="https://dplyr.tidyverse.org/reference/between.html" class="external-link">between()</a></code>, <code><a href="https://dplyr.tidyverse.org/reference/near.html" class="external-link">near()</a></code></p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://vroom.r-lib.org/reference/vroom.html" class="external-link">vroom::vroom()</a></code> <code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Get path to example file</span></span></span>
<span class="r-in"><span><span class="va">ref_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="miplicorn_example.html">miplicorn_example</a></span><span class="op">(</span><span class="st">"reference_AA_table.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">alt_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="miplicorn_example.html">miplicorn_example</a></span><span class="op">(</span><span class="st">"alternate_AA_table.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cov_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="miplicorn_example.html">miplicorn_example</a></span><span class="op">(</span><span class="st">"coverage_AA_table.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ref_file</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/home/runner/work/_temp/Library/miplicorn/extdata/reference_AA_table.csv"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Input sources -------------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="co"># Read from a path</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_reference</span><span class="op">(</span><span class="va">ref_file</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A reference table: 6,344 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ ref_u…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 6,334 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​ref_umi_count</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># You can also use paths directly</span></span></span>
<span class="r-in"><span><span class="co"># read_tbl_alternate("alternate_AA_table.csv")</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Read entire file ----------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_coverage</span><span class="op">(</span><span class="va">cov_file</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A coverage table: 6,344 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ cover…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 6,334 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​coverage</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Data filtering ------------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="co"># Filtering by one criterion</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_reference</span><span class="op">(</span><span class="va">ref_file</span>, <span class="va">gene</span> <span class="op">==</span> <span class="st">"atp6"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A reference table: 260 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ ref_u…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 250 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​ref_umi_count</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filtering by multiple criteria within a single logical expression</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_alternate</span><span class="op">(</span><span class="va">alt_file</span>, <span class="va">gene</span> <span class="op">==</span> <span class="st">"atp6"</span> <span class="op">&amp;</span> <span class="va">targeted</span> <span class="op">==</span> <span class="st">"Yes"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># An alternate table: 156 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ alt_u…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 146 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​alt_umi_count</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span><span class="fu">read_tbl_coverage</span><span class="op">(</span><span class="va">cov_file</span>, <span class="va">gene</span> <span class="op">==</span> <span class="st">"atp6"</span> <span class="op">|</span> <span class="va">targeted</span> <span class="op">==</span> <span class="st">"Yes"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A coverage table: 2,496 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ cover…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 2,486 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​coverage</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># When multiple expressions are used, they are combined using &amp;</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_reference</span><span class="op">(</span><span class="va">ref_file</span>, <span class="va">gene</span> <span class="op">==</span> <span class="st">"atp6"</span>, <span class="va">targeted</span> <span class="op">==</span> <span class="st">"Yes"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A reference table: 156 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample     gene_id       gene  mutation_name  exoni…¹ aa_ch…² targe…³ ref_u…⁴</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes          90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missen… Ala623… Yes         175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 146 more rows, and abbreviated variable names ¹​exonic_func,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​aa_change, ³​targeted, ⁴​ref_umi_count</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Read multiple files together ----------------------------------------------</span></span></span>
<span class="r-in"><span><span class="fu">read_tbl_ref_alt_cov</span><span class="op">(</span><span class="va">ref_file</span>, <span class="va">alt_file</span>, <span class="va">cov_file</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A ref alt cov table: 6,344 × 10</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample  gene_id gene  mutat…¹ exoni…² aa_ch…³ targe…⁴ ref_u…⁵ alt_u…⁶ cover…⁷</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes         608       0     608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes          20       0      20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes         158       0     158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes           2       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes           1       0       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes         129       0     129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes           0       0       0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes           0       0       0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes          90       0      90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> D10-JJ… PF3D7_… atp6  atp6-A… missen… Ala623… Yes         175       0     175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 6,334 more rows, and abbreviated variable names ¹​mutation_name,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​exonic_func, ³​aa_change, ⁴​targeted, ⁵​ref_umi_count, ⁶​alt_umi_count,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ⁷​coverage</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://www.arispas.com/" class="external-link">Aris Paschalidis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

